### DISCUSSION TITLE
Non product and collection metafields?

### DISCUSSION THREAD

**Discussion by dungle-scrubs** [gatsbyjs/gatsby#32091] [Help]
Is there already, or will there at some point be access to all types of metafields?  For example, I use an app called Custom Fields by Bonify and they offer a global metafield section.  I like it because it allows for arrays of product or metafield references (JSON stringified array of handles) which are really simple for the client.

Searching through `allShopifyMetafieldInterface`, I don't see the global metafields I've created, so am just wondering if they are limited to products and collections for now.

As a use-case, I think globals would be really nice for certain types of site configuration with headless shops.  

Cheers!

**Comment by sslotsky** [comment]
Do you know if these global metafields are available through the admin API?

We currently pull down metafields for products, product variants, and collections. Within Gatsby these each have their own types, but they also implement a common interface which is what allows them to be retrieved by `allShopifyMetafieldInterface`. So if there are other resources that have metafields, we could pull those down as well and add them to the interface. But I'm not currently seeing anything in the admin API that allows us to pull down these global metafields (I might be missing it though)

**Comment by dungle-scrubs** [comment]
@sslotsky Ah okay, thanks, that makes sense.  I'm assuming that the metafields I'm referring to are available through the Admin API as `Shop.metafields`, which this plugin isn't pulling in.

From the docs, the complete list of Types that return metafields are

Collection.metafield
Collection.metafields
Customer.metafield
Customer.metafields
DraftOrder.metafield
DraftOrder.metafields
Image.metafield
Image.metafields
Order.metafield
Order.metafields
Product.metafield
Product.metafields
ProductVariant.metafield
ProductVariant.metafields
Shop.metafield
Shop.metafields

**Reply by hudsonandtask** [reply]
Hi do you still think this is the case? Does this plugin not pull these other metafield types?

**Comment by sslotsky** [comment]
I'll have to double check this but my recollection is that we couldn't get the `shop` through the bulk API because it only allowed us to request connection data at the top level. But maybe there are other ways to get at those `shop.Metafields`. Thanks for assembling this information.

**Comment by dungle-scrubs** [comment]
You're welcome.  And the Admin API can still be pinged separately in `gatsby-node`.

**Comment by sslotsky** [comment]
It certainly can. We are just cautious about having more than one way to source the data without at least understanding your use cases first. The logic to consume the regular admin API vs the bulk API is quite different.

**Comment by dungle-scrubs** [comment]
Thanks, I think my use-case is way too specific to make that worthwhile.  In most cases with a headless shop, shop-owner-configurable site-wide settings can be handled via a CMS.
